src/Microsoft.AspNetCore.Razor.Tools/ShutdownCommand.cs

```diff
@@ -61,11 +62,14 @@ protected async override Task<int> ExecuteCoreAsync()
                         try
                         {
                             var process = Process.GetProcessById(response.ServerProcessId);
-                            process.WaitForExit();
+
+                            // Sometimes in linux when the process has already exited before we get here, WaitForExit tends to wait indefinitely.
+                            // Add a timeout to avoid hang.
+                            process.WaitForExit(TimeoutMsProcessShutdown);
```

Comment:
Should we report this issue on CoreFX? The windows behaviour is to throw if the process has exited. 

If this is really the case, perhaps we should do this in a loop and check if the process is still running with a shorter timeout?

Reply:
👍 for filing a CoreFx issue, at the very least so that other users of the API could see it.

The test code has a slightly different pattern for waiting for process exit. Should we use that here - https://github.com/aspnet/Razor/blob/3e436a53a1a0555e05111594be96323d16cc88b8/test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildProcessManager.cs#L83-L108?